jmdaemon/jmdaemon/onionmc.py

```diff
@@ -0,0 +1,1210 @@
+from jmdaemon.message_channel import MessageChannel
+from jmdaemon.protocol import COMMAND_PREFIX, JM_VERSION
+from jmbase import get_log,  JM_APP_NAME, JMHiddenService
+import json
+import copy
+from typing import Callable, Union, Tuple, List
+from twisted.internet import reactor, task, protocol
+from twisted.protocols import basic
+from twisted.internet.endpoints import TCP4ClientEndpoint
+from twisted.internet.address import IPv4Address, IPv6Address
+from txtorcon.socks import TorSocksEndpoint
+
+log = get_log()
+
+
+NOT_SERVING_ONION_HOSTNAME = "NOT-SERVING-ONION"
+
+def location_tuple_to_str(t: Tuple[str, int]):
+    return t[0] + ":" + str(t[1])
+
+def is_non_serving_location(l: str):
+    return l == NOT_SERVING_ONION_HOSTNAME
+
+def is_local_location(l: str) -> bool:
+    try:
+        a, b = l.split(":")
+    except:
+        return False
+    return a in ["localhost", "127.0.0.1"]

```

Comment:
Well, it certainly seems reasonable, I was quite uncomfortable writing this function as it should definitely be delegated to something else, but I wasn't sure what. Your version looks fine to me.

Reply:
I just looked at this again, it raises the important point that 'don't send if it's local' is a crappy way to distinguish an inbound connection, since the logic doesn't work in test (*everything* is local, there ...). Will need to look at this separately.

Reply:
Fixed in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1182/commits/474dfa0514ac62ca263555b45a2e5888006cbf86 ; this function is now removed.